app/models/metric/targets.rb

```diff
@@ -19,7 +19,8 @@ def self.capture_infra_targets(zone, options = {})
     targets += zone.storages.select { |s| Storage::SUPPORTED_STORAGE_TYPES.include?(s.store_type) } unless options[:exclude_storages]
 
     targets = targets.select(&:perf_capture_enabled?)
-    targets = targets.collect { |t| t.kind_of?(EmsCluster) ? t.hosts : t }.flatten
+    MiqPreloader.preload(targets.select { |t| t.kind_of?(EmsCluster) }, :hosts)
```

Comment:
This is the N+1 fix?  Would It be simpler to just adjust the includes on line 14 to `:ems_clusters => [:hosts, :tags]`?